src/shims/foreign_items/posix.rs

```diff
@@ -136,6 +136,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // "lseek" is only used on macOS which is 64bit-only, so `i64` always works.
                 this.write_scalar(Scalar::from_i64(result), dest)?;
             }
+            "fsync" => {
+                let result = this.fsync(args[0])?;
```

Comment:
Please use `check_arg_count` to extract the arguments, like all other shims do now.